### PR TITLE
Replace microprofile-config impl with smallrye-config.

### PR DESCRIPTION
**Pull request by bobmcwhirter**
- Adjust for lack of Optional<Collection<T>> in mp-config.
- Provide ConfigSourceProviders for extra functionality.

Fixes https://github.com/thorntail/thorntail/issues/967

- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [ ] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [x] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -90,7 +93,7 @@
     <version.commons-logging>1.2</version.commons-logging>
     <version.netty>4.1.24.Final</version.netty>
 
-    <version.mp-config>1.2.1</version.mp-config>
+    <version.mp-config>1.3</version.mp-config>
```
Comment (kenfinnigan) [inline bom/pom.xml:96]: Just wondering, do we need the MP Config API in the build as well as SmallRye Config?

Comment (bobmcwhirter) [reply bom/pom.xml:96]: Nope, in fact we don't. I removed the <dependency> but missed the property, I think.

**Comment by sberyozkin** [comment]
the fault-tolerance test is failing

**Comment by bobmcwhirter** [comment]
Hrm, it passes locally. I'll double-check.

**Comment by sberyozkin** [comment]
Hi Bob, not sure if it is related but while I was looking at the keycloak test I ended up lowering the the event priority in one of the customizers to ensure the ordering is predictable, see #1009

**Comment by sberyozkin** [comment]
This branch has few conflicts now

**Comment by bobmcwhirter** [comment]
I’ll rebase

**Comment by bobmcwhirter** [comment]
Let's see how that rebase goes. Passing locally now.

File: bom/pom.xml (hunk 1 above)
Comment (bobmcwhirter) [reply bom/pom.xml:96]: We do need the `version.mp-config` to reference the TCK, and in that case, I think keeping the TCK and API in sync within the context of Thorntail might be smart?

Comment (kenfinnigan) [reply bom/pom.xml:96]: is there a way to ensure one doesn't get updated without the other?

trying to minimize moving pieces

Comment (bobmcwhirter) [reply bom/pom.xml:96]: I think at this point, not easily. We can keep our API + TCK in sync, but then SmallRye might vary.  We can keep SmallRye + API in sync, but then TCK might vary.

**Review by sberyozkin — APPROVED** [review]
Ready to go, though the 4.x line is currently broken

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**